src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1916 @@
+
+            if (ShouldConsumeSequenceElement(consumeCloseParen))
+            {
+                do
+                {
+                    var last = list.Count == 0 ? null : list.Last();
+                    list.Add(ParsePrimaryExpressionAndQuantifiers(last));
+
+                    TryMergeLastTwoNodes(list);
+                }
+                while (ShouldConsumeSequenceElement(consumeCloseParen));
+            }
+
+            return new RegexSequenceNode(list.ToImmutableAndFree());
+        }
+
+        private void TryMergeLastTwoNodes(ArrayBuilder<RegexExpressionNode> list)
+        {
+            if (list.Count >= 2)
+            {
+                var last = list[list.Count - 2];
+                var next = list[list.Count - 1];
+
+                if (last?.Kind == RegexKind.Text && next?.Kind == RegexKind.Text)
+                {
+                    var lastTextToken = ((RegexTextNode)last).TextToken;
+                    var nextTextToken = ((RegexTextNode)next).TextToken;
+
+                    if (lastTextToken.Diagnostics.Length == 0 &&
+                        nextTextToken.Diagnostics.Length == 0 &&
+                        lastTextToken.Value == null &&
+                        nextTextToken.Value == null &&
+                        nextTextToken.LeadingTrivia.Length == 0)
+                    {
+                        // Merge two text tokens token if there is no intermediary trivia.
+                        var merged = new RegexTextNode(CreateToken(
+                            RegexKind.TextToken, lastTextToken.LeadingTrivia, 
+                            lastTextToken.VirtualChars.Concat(nextTextToken.VirtualChars)));
+
+                        list.RemoveLast();
+                        list.RemoveLast();
+                        list.Add(merged);
+                    }
+                }
+            }
+        }
+
+        private bool ShouldConsumeSequenceElement(bool consumeCloseParen)
+        {
+            if (_currentToken.Kind == RegexKind.EndOfFile)
+            {
+                return false;
+            }
+
+            if (_currentToken.Kind == RegexKind.BarToken)
+            {
+                return false;
+            }
+
+            if (_currentToken.Kind == RegexKind.CloseParenToken)
+            {
+                return consumeCloseParen;
+            }
+
+            return true;
+        }
+
+        private RegexExpressionNode ParsePrimaryExpressionAndQuantifiers(RegexExpressionNode lastExpression)
+        {
+            var current = ParsePrimaryExpression(lastExpression);
+            if (current.Kind == RegexKind.SimpleOptionsGrouping)
+            {
+                // Simple options (i.e. "(?i-x)" can't have quantifiers attached to them).
+                return current;
+            }
+
+            switch (_currentToken.Kind)
+            {
+                case RegexKind.AsteriskToken: return ParseZeroOrMoreQuantifier(current);
```

Comment:
Maybe fall through for the first 3 cases?

Reply:
#won'tfix